### DISCUSSION TITLE
WarpX Install Question: ADIOS2 on HPC3 (UCI)

### DISCUSSION THREAD

**Discussion by jinze-liu** [BLAST-WarpX/warpx#4832] [Installation] [answered]
Thank you for the community's responses which have resolved many of my questions and saved me a lot of time. However, I still have a few questions I'd like to ask:

When I attempt to compile from source, my HPC does not match any of the provided examples. My HPC now includes an NVIDIA A100 GPU. I modified and configured according to the HPC3 (UCI) example document.

I modified the example document of HPC3(UCI)  to configure my environment, and the environment should have been installed：
```
#!/bin/bash
export proj="ljz_gpu"
export MY_PROFILE=$(cd $(dirname $BASH_SOURCE) && pwd)/$(basename $BASH_SOURCE)
module load gcc/9.4.0-gcc-4.8.5   
module load cmake/3.25.2-gcc-9.4.0  
module load cuda/11.8.0-gcc-4.8.5  
module load openmpi/4.1.5-gcc-9.4.0  
module load boost/1.80.0-gcc-9.4.0    
# optional: for openPMD and PSATD+RZ support
module load openblas/0.3.21-gcc-9.4.0
module load  hdf5/1.14.0-gcc-9.4.0  
export CMAKE_PREFIX_PATH=${HOME}/sw/hpc3/gpu/c-blosc-1.21.1:$CMAKE_PREFIX_PATH
export CMAKE_PREFIX_PATH=${HOME}/sw/hpc3/gpu/adios2-2.8.3:$CMAKE_PREFIX_PATH
export CMAKE_PREFIX_PATH=${HOME}/sw/hpc3/gpu/blaspp-master:$CMAKE_PREFIX_PATH
export CMAKE_PREFIX_PATH=${HOME}/sw/hpc3/gpu/lapackpp-master:$CMAKE_PREFIX_PATH
export LD_LIBRARY_PATH=${HOME}/sw/hpc3/gpu/c-blosc-1.21.1/lib64:$LD_LIBRARY_PATH
export LD_LIBRARY_PATH=${HOME}/sw/hpc3/gpu/adios2-2.8.3/lib64:$LD_LIBRARY_PATH
export LD_LIBRARY_PATH=${HOME}/sw/hpc3/gpu/blaspp-master/lib64:$LD_LIBRARY_PATH
export LD_LIBRARY_PATH=${HOME}/sw/hpc3/gpu/lapackpp-master/lib64:$LD_LIBRARY_PATH
export PATH=${HOME}/sw/hpc3/gpu/adios2-2.8.3/bin:${PATH}
module load python/3.10.6-gcc-4.8.5  
if [ -d "${HOME}/sw/hpc3/gpu/venvs/warpx-gpu" ]
then
  source ${HOME}/sw/hpc3/gpu/venvs/warpx-gpu/bin/activate
fi
# an alias to request an interactive batch node for one hour
#   for parallel execution, start on the batch node: srun <command>
alias getNode="salloc -N 1 -t 0:30:00 --gres=gpu:A100:1 -p free-gpu"
# an alias to run a command on a batch node for up to 30min
#   usage: runNode <command>
alias runNode="srun -N 1 -t 0:30:00 --gres=gpu:A100:1 -p free-gpu"
export AMREX_CUDA_ARCH=8.0
# compiler environment hints
export CXX=$(which g++)
export CC=$(which gcc)
export FC=$(which gfortran)
export CUDACXX=$(which nvcc)
export CUDAHOSTCXX=${CXX}
```

 However, when I run the command 'bash $HOME/src/warpx/Tools/machines/hpc3-uci/install_gpu_dependencies.sh', it throws an error. The error is attached below in the form of an image.
![1](https://github.com/ECP-WarpX/WarpX/assets/134666994/4374f725-ca41-4c88-a784-b2b2116765db)

I am quite new to this, so my questions might be both simple and numerous. Thanks for your patience and answers!

**Comment by ax3l** [comment, marked as answer]
Oh, that is an incompatibility between a specific ADIOS2 and a specific HDF5 version. This PR will fix it, please update the respective `Tools/machines/hpc3-uci/install_gpu_dependencies.sh` file: https://github.com/ECP-WarpX/WarpX/pull/4836

Can you report in the PR if that fixes it for you?
